pkg/cloudprovider/providers/vsphere/vclib/virtualmachine.go

```diff
@@ -202,6 +202,27 @@ func (vm *VirtualMachine) IsActive(ctx context.Context) (bool, error) {
 	return false, nil
 }
 
+// Exists checks if VM exists and is not terminated
+func (vm *VirtualMachine) Exists(ctx context.Context) (bool, error) {
+	vmMoList, err := vm.Datacenter.GetVMMoList(ctx, []*VirtualMachine{vm}, []string{"summary"})
+	if err != nil {
+		glog.Errorf("Failed to get VM Managed object with property summary. err: +%v", err)
+		return false, err
+	}
+	aliveStates := []types.VirtualMachinePowerState{
```

Comment:
`PowerState` can only be one of these 3 values, is there a need to check the state value at all?  Or have you seen a VM somehow in another power state?

Reply:
ok, I suppose it is fine as-is, could you add a comment to clarify for others?